### PR TITLE
update generate yaml file

### PR DESCRIPTION
**Pull request by jichenjc**
**What this PR does / why we need it**:

**Which issue(s) this PR fixes** *(optional, in `fixes #<issue number>(, fixes #<issue_number>, ...)` format, will close the issue(s) when PR gets merged)*:
Fixes #

**Special notes for your reviewer**:

1. Please confirm that if this PR changes any image versions, then that's the sole change this PR makes.

**Release note**:

```release-note

```

### CODE REVIEW

**Comment by jichenjc** [comment]
@xunpan can you take a look at this?

after this patch, we can start to debug create functions such as 

../../../clusterctl create cluster --bootstrap-type kind  --provider ibmcloud  -c cluster.yaml -m machines.yaml -p provider-components.yaml -v 10

File: cmd/clusterctl/configuration/ibmcloud/generate-yaml.sh [hunk 1]
```diff
@@ -1,3 +1,124 @@
...
+# Set up the output dir if it does not yet exist
+mkdir -p $PWD/out
+cp -n $PWD/cluster.yaml.template $PWD/out/cluster.yaml
+cp -n $PWD/machines.yaml.template $PWD/out/machines.yaml
+
+OS=$(uname)
+if [[ "$OS" =~ "Linux" ]]; then
+  OPENSTACK_CLOUD_PROVIDER_CONF=$(echo "$OPENSTACK_CLOUD_PROVIDER_CONF_PLAIN"|base64 -w0)
+elif [[ "$OS" =~ "Darwin" ]]; then
+  OPENSTACK_CLOUD_PROVIDER_CONF=$(echo "$OPENSTACK_CLOUD_PROVIDER_CONF_PLAIN"|base64)
```
Comment (xunpan) [inline cmd/clusterctl/configuration/ibmcloud/generate-yaml.sh]: This `OPENSTACK_CLOUD_PROVIDER_CONF` related code seems useless.

File: cmd/clusterctl/configuration/ibmcloud/generate-yaml.sh [hunk 2]
```diff
@@ -1,3 +1,124 @@
...
+else
+  echo "Unrecognized OS : $OS"
+  exit 1
+fi
+
+# Build provider-components.yaml with kustomize
+#echo "---" >> $PWD/out/provider-components.yaml
+#kustomize build $PWD/provider-component/clouds-secrets >> $PWD/out/provider-components.yaml
```
Comment (xunpan) [inline cmd/clusterctl/configuration/ibmcloud/generate-yaml.sh]: Do you have plan to support it? It is mandatory for cloud operation. If it does not in your work list, I can do it during test `Create` API of `Machine` controller.

**Comment by xunpan** [comment]
Another commits, could you please squash the commits or at least make `generate-yaml.sh` with no add/remove lines in 2 comments.

File: cmd/clusterctl/configuration/ibmcloud/generate-yaml.sh (hunk 1 above)
Comment (jichenjc) [reply cmd/clusterctl/configuration/ibmcloud/generate-yaml.sh]: ok

File: cmd/clusterctl/configuration/ibmcloud/generate-yaml.sh (hunk 2 above)
Comment (jichenjc) [reply cmd/clusterctl/configuration/ibmcloud/generate-yaml.sh]: ok, I will add this first

File: cmd/clusterctl/configuration/ibmcloud/generate-yaml.sh [hunk 3]
```diff
@@ -1,3 +1,119 @@
-cat clouds.yaml.template  > clouds.yaml
-cat cluster.yaml.template > cluster.yaml
-cat  machines.yaml.template > machines.yaml
+/!/bin/bash
```
Comment (xunpan) [inline cmd/clusterctl/configuration/ibmcloud/generate-yaml.sh]: I guess it is a typo.

**Comment by xunpan** [comment]
This is a good PR to connect whole cluster creation process. Thanks!

**Pull request merged**